libqtile/widget/tvheadend.py

```diff
@@ -0,0 +1,327 @@
+# Copyright (c) 2020 elParaguayo
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import os
+from datetime import datetime
+from pathlib import Path
+
+import requests
+from requests.auth import HTTPBasicAuth
+
+from libqtile import bar, images
+from libqtile.log_utils import logger
+from libqtile.popup import Popup
+from libqtile.widget import base
+
+UPCOMING_RECORDINGS = "/dvr/entry/grid_upcoming"
+
+
+def icon_path():
+    """Get the path to tv icon"""
+    dir_path = Path(__file__).resolve() / ".." / ".." / "resources" / "tvheadend-icons"
+    return str(dir_path.resolve())
+
+
+class TVHTimeOut(Exception):
+    pass
+
+
+class TVHJobServer(object):
+
+    def __init__(self, host=None, auth=None, timeout=5):
+        self.host = host
+        self.auth = auth
+        self.timeout = timeout
+
+    def _send_api_request(self, path, args=None):
+        url = self.host + path
+        try:
+            r = requests.post(url,
+                              data=args,
+                              auth=self.auth,
+                              timeout=self.timeout)
+            return r.json(strict=False)
+        except (requests.exceptions.Timeout, requests.ConnectionError):
+            raise TVHTimeOut()
```

Comment:
This re-raising swallows the cause for the original exception, can we just drop this?